pkg/actuators/client/client.go

```diff
@@ -20,86 +20,407 @@ import (
+	resourceManagerService, err := resourcemanagerv2.NewResourceManagerV2(&resourcemanagerv2.ResourceManagerV2Options{
+		Authenticator: authenticator,
+	})
+	if err != nil {
+		return nil, err
 	}
+
 	return &ibmCloudClient{
-		vpcService: vpcService,
+		vpcService:             vpcService,
+		resourceManagerService: resourceManagerService,
 	}, nil
 }
 
-// InstanceGet returns retrieves a single instance specified by instanceID
-func (c *ibmCloudClient) InstanceGet(instanceID string) (*vpcv1.Instance, error) {
+// InstanceExistsByName checks if the instance exist in VPC
+func (c *ibmCloudClient) InstanceExistsByName(name string, machineProviderConfig *ibmcloudproviderv1.IBMCloudMachineProviderSpec) (bool, error) {
+	// Get Instance info
+	_, err := c.InstanceGetByName(name, machineProviderConfig)
+
+	// Instance found
+	if err == nil {
+		return true, nil
+	}
+
+	// Instance not found
+	if err.Error() == "Instance not found" {
+		return false, nil
+	}
+
+	// Could not retrieve Instances list
+	return false, err
+}
+
+// InstanceDeleteByName deletes the requested instance
+func (c *ibmCloudClient) InstanceDeleteByName(name string, machineProviderConfig *ibmcloudproviderv1.IBMCloudMachineProviderSpec) error {
+	// Get Instance info
+	getInstance, err := c.InstanceGetByName(name, machineProviderConfig)
+	if err != nil {
+		return err
+	}
+
+	// Get instance ID
+	instanceID := *getInstance.ID
+	if instanceID == "" {
+		return fmt.Errorf("Could not get the Instance ID")
+	}
+
+	// Initialize New Delete Instance Options
+	deleteInstanceOption := c.vpcService.NewDeleteInstanceOptions(instanceID)
+	// // Set Instance ID
+	// deleteInstanceOption.SetID(instanceID)
+
+	// Delete the Instance
+	_, err = c.vpcService.DeleteInstance(deleteInstanceOption)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// InstanceGetByName retrieves a single instance specified by Instance Name
+func (c *ibmCloudClient) InstanceGetByName(name string, machineProviderConfig *ibmcloudproviderv1.IBMCloudMachineProviderSpec) (*vpcv1.Instance, error) {
+	// Get region info
+	regionName := machineProviderConfig.Region
+	region, _, err := c.vpcService.GetRegion(c.vpcService.NewGetRegionOptions(regionName))
+	if err != nil {
+		return nil, err
+	}
+
+	// Set the Service URL
+	err = c.vpcService.SetServiceURL(fmt.Sprintf("%s/v1", *region.Endpoint))
+	if err != nil {
+		return nil, err
+	}
+
+	// Get Service URL
+	serviceURL := c.vpcService.GetServiceURL()
+	// Initialize New List Instances Options
+	listInstOptions := c.vpcService.NewListInstancesOptions()
+	// Set Image Name
+	listInstOptions.SetName(name)
+	// Set VPC Name
+	vpcName := machineProviderConfig.VPC
+	listInstOptions.SetVPCName(vpcName)
+
+	// Get Instances list
+	instance, _, err := c.vpcService.ListInstances(listInstOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check if instance is not nil
+	if instance != nil {
+		for _, eachInstance := range instance.Instances {
+			if name == *eachInstance.Name {
```

Comment:
With the `name` filter being designated in the request options, we shouldn't need to check for a name match here.
According to the sdk doc, the `name` parameter:
> "Filters the collection to resources with the exact specified name."

Reply:
That was my initial thought as well.

I left the name check in there just to double-check. It should not be needed, but after working with the SDK, I thought there is no harm to have that extra check. 

...should be an easy fix, if we want to do that?

Reply:
Up to you Gagan - just pointing out that it shouldn't be necessary.